proxy/pooldata.lua

```diff
@@ -0,0 +1,25 @@
+-- if my_zone() is commented out, will look for "backends" instead of "zones"
+my_zone("z1")
+
+-- NOTE: optional to specify this if defaults are okay.
+router{
+    router_type = "keyprefix",
+	match_prefix = "/(%a+)/",
+	default_pool = nil
+}
+
+-- NOTE: a normal config will have backends _or_ zones, not both!
+pool{
+	name = "foo",
+	backends = {"127.0.0.1:11212", "127.0.0.1:11213"},
+	zones = {
+      z1 = {
```

Comment:
1. Could document that this is (I assume) meant to be a DSL but internally a function call with optional parenthesis with a table as an argument https://stackoverflow.com/questions/36126249/no-parentheses-after-a-function-name/36126497 - people completely unfamiliar with lua could stumble over that or not be sure how to customize the config passed to a `pool`, e.g. in a loop or with optional fields
2. nit: This has a mix of 2 spaces and tabs for indentation. https://editorconfig.org/ may help avoid this in the future for contributions from others for `[*.lua]`, c, etc., or run `expand`/`unexpand` - I haven't seen it suggested before on the github PRs/issues but may have missed it.

Reply:
Yeah; as I document this I'll try to make that obvious. A lot of people don't want to think even that far, so I'd present it as is then break it down a bit. I'm planning an doing an introductory blog post around this simple interface, then do a series around more detailed examples... and building out the wiki docs from those. Need to build out the simple interface a bit more today and tomorrow.

Sorry my config's set for expandtab so not sure how that happened (possibly missing for lua? I'll figure it out later). I noticed it after pasting bu wast too lazy to fix.